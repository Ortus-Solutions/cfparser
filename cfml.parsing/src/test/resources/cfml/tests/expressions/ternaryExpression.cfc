s = a && b?"a":"b";