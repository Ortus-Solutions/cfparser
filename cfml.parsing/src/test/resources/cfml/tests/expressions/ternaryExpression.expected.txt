/*===TOKENS===*/
IDENTIFIER               <s>
'='                      <=>
IDENTIFIER               <a>
'&&'                     <&&>
IDENTIFIER               <b>
'?'                      <?>
OPEN_STRING              <">
STRING_LITERAL           <a>
CLOSE_STRING             <">
':'                      <:>
OPEN_STRING              <">
STRING_LITERAL           <b>
CLOSE_STRING             <">
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (assignmentExpression
        (startExpression
          (baseExpression (unaryExpression (memberExpression (identifier s))))
        )
        =
        (startExpression
          (baseExpression
            (baseExpression
              (baseExpression (unaryExpression (memberExpression (identifier a))))
              &&
              (baseExpression (unaryExpression (memberExpression (identifier b))))
            )
            (ternaryExpression
              ?
              (startExpression
                (baseExpression
                  (unaryExpression
                    (primaryExpression
                      (literalExpression (stringLiteral " (stringLiteralPart a) "))
                    )
                  )
                )
              )
              :
              (startExpression
                (baseExpression
                  (unaryExpression
                    (primaryExpression
                      (literalExpression (stringLiteral " (stringLiteralPart b) "))
                    )
                  )
                )
              )
            )
          )
        )
      )
      (endOfStatement ;)
    )
  )
)
/*======*/
/*===DECOMPILE===*/
s = a && b?'a':'b'
/*======*/